src/FlatbufferMessage.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include "KafkaW/Msg.h"
+#include "logger.h"
+#include <atomic>
+#include <cstddef>
+#include <cstdint>
+#include <librdkafka/rdkafka.h>
+#include <librdkafka/rdkafkacpp.h>
+#include <memory>
+#include <vector>
+
+namespace FileWriter {
+class BufferTooSmallError : public std::runtime_error {
+public:
+  explicit BufferTooSmallError(const std::string &what)
+      : std::runtime_error(what){};
+};
+
+class UnknownFlatbufferID : public std::runtime_error {
+public:
+  explicit UnknownFlatbufferID(const std::string &what)
+      : std::runtime_error(what){};
+};
+
+class NotValidFlatbuffer : public std::runtime_error {
+public:
+  explicit NotValidFlatbuffer(const std::string &what)
+      : std::runtime_error(what){};
+};
+/// \todo Profile code to determine if the exceptions are a performance problem
+
+/// A wrapper around a databuffer which holds a flatbuffer. Used to simplify
+/// passing around flatbuffers and the most important pieces of meta-data.
+/// \note Does not take ownership of any pointers. You must make sure to
+/// free the pointers you passed yourself.
+class FlatbufferMessage {
+public:
+  /// Constructor is used in unit testing code to simplify set-up.
+  FlatbufferMessage() = default;
+  /// Verifies the data in the flatbuffer to make sure if it is valid.
+  /// \param[in] BufferPtr Pointer to memory location containing flatbuffer.
+  /// \param[in] Size Size of flatbuffer in bytes.
+  /// \throws FileWriter::BufferTooSmallError If the size argument is < 8, it
+  /// can not be a flatbuffer as the identifier is stored in bytes 4 to 8.
+  /// \throws FileWriter::UnknownFlatbufferID If the flatbuffer ID (bytes 4 to
+  /// 8) is not found in the database, this exception is thrown.
+  /// \throws FileWriter::NotValidFlatbuffer The constructor runs
+  /// FileWriter::FlatbufferReader::verify() on the flatbuffer and this
+  /// exception is thrown if it returns false.
+  FlatbufferMessage(char const *BufferPtr, size_t const Size);
```

Comment:
The `const` for `Size`has no effect here in the function declaration.

Reply:
That is one of the most esoteric rules in C++ that I have encountered. Do you have a reference to why this is so? Regardless, the rule is way too esoteric for me to change the code.

Reply:
In this case the `const`ness is an implementation detail which is irrelevant to the interface, on the other hand if that seems odd to people then that is reason enough to leave `const` in the declaration.

Reply:
I understand that this is the reason why `const` is not required. What makes this surprising to me is because having the interface reflect the implementation requires only one rule in the C++ standard whereas the actual behaviour requires two rules in the standard.

Anyway, I am willing to change the code if you can find one more person that agrees with you.